lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +43,21 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or text == '':
+            return None
+        no_letters = 0
+        for i in text:
+            if not i.isalpha():
+                no_letters += 1
+                if len(text) == no_letters:
+                    return None
```

Comment:
You could have checked it at the very end: if the text consists of digits and spaces, then the resulting tokenized list would be empty, right?